### PR TITLE
adding callback structure to couple with callbacks on non-datasource …

### PR DESCRIPTION
**Pull request by Stasik0**
…nodes, relates to #399

@ichrispa I changed the signature of UA_Server_setAttribute_Datasource a bit
@jpfr review please
@tpiat this might interest you

### CODE REVIEW

File: include/ua_server.h [hunk 1]
```diff
@@ -119,6 +119,12 @@ typedef struct {
     UA_StatusCode (*write)(void *handle, const UA_NodeId nodeid, const UA_Variant *data, const UA_NumericRange *range);
 } UA_DataSource;
 
+typedef struct {
+    void *handle;
+    void (*onRead)(void *handle, const UA_NodeId nodeid,  const UA_Variant *data, const UA_NumericRange *range);
+    void (*onWrite)(void *handle, const UA_NodeId nodeid, const UA_Variant *data, const UA_NumericRange *range);
+} UA_NotificationCallback;
```
Comment (jpfr) [inline include/ua_server.h]: NotificationCallback has an ambiguous semantics.
You want to callback into userspace. Not the internal notificaiton/subscription layer.

Comment (Stasik0) [inline include/ua_server.h]: Propose a better name!

Comment (Stasik0) [inline include/ua_server.h]: UA_UserspaceCallback?

**Comment by Stasik0** [comment]
+1 anyone?

**Pull request merged**